src/anet.c

```diff
@@ -494,7 +495,13 @@ int anetUnixServer(char *err, char *path, mode_t perm, int backlog)
 static int anetGenericAccept(char *err, int s, struct sockaddr *sa, socklen_t *len) {
     int fd;
     do {
+        /* Use the accept4() call on linux to simultaneously accept and
+         * set a socket as non-blocking. */
+#ifdef HAVE_ACCEPT4
+        fd = accept4(s, sa, len,  SOCK_NONBLOCK | SOCK_CLOEXEC);
+#else
         fd = accept(s,sa,len);
+#endif
```

Comment:
This makes the function return a non-blocking socket on Linux and a blocking socket on other OSes.

Can we do `anetNonBlock(NULL, fd)` and `anetCloexec(fd)` in the `#else` branch here, so we have the same socket flags here regardless of OS?

Reply:
good idea... it was bothering me too (that we have coupling between networking.c and anet.c, both use the same ifdef and the fact networking.c relies on some internal behavior of anet.c), but i was lazy and decided not to act on it..
i'll make the change.